File: plugin1/src/main/java/com/luoyunmin/plugin1/TimePlugin.java
package com.luoyunmin.plugin1;

import org.gradle.api.Action;
import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.Task;

public class TimePlugin implements Plugin<Project> {
    @Override
    public void apply(Project target) {
        target.task("plugin1TimePlugin").doLast(new Action<Task>() {
            @Override
            public void execute(Task task) {
                System.out.println("this is plugin1TimePlugin!");
            }
        });
    }
}

File: plugin1/src/main/java/com/luoyunmin/plugin1/TimePlugin.kt
package com.luoyunmin.plugin1

import com.android.build.gradle.AppExtension
import org.gradle.api.Plugin
import org.gradle.api.Project

class TimePlugin : Plugin<Project> {
    override fun apply(project: Project) {
        println("this is TimePlugin!")
        val appExtension = project.extensions.findByType(AppExtension::class.java)
        appExtension?.registerTransform(TimeTransform())
    }
}